libs/framework/include/celix/dm/Component.h

```diff
@@ -232,17 +232,29 @@ namespace celix { namespace dm {
          * @return the DM Component reference for chaining (fluent API)
          */
         Component<T>& setCallbacks(
-            int (T::*init)(),
-            int (T::*start)(),
-            int (T::*stop)(),
-            int (T::*deinit)()
+                int (T::*init)(),
+                int (T::*start)(),
+                int (T::*stop)(),
+                int (T::*deinit)()
         );
         /**
          * Remove the previously registered callbacks for the component life cycle control
          *
          * @return the DM Component reference for chaining (fluent API)
          */
         Component<T>& removeCallbacks();
+
+        /**
+         * Safely copies a std::string into a char* buffer
+         * @param input string to copy
+         * @param dst buffer to allocate into, owner is responsible for freeing memory
+         */
+        void copyString(const std::string& input, char **dst)
```

Comment:
Should this be used for all c_str calls?

Reply:
Everywhere ownership is transferred yes, but now that I think about it, the properties_set function copies and doesn't transfer ownership. Therefore it is not needed here at all.